lib/thor/actions/file_manipulation.rb

```diff
@@ -340,7 +340,8 @@ def capture(*args)
       with_output_buffer { yield(*args) }
     end
 
-    def with_output_buffer(buf = "") #:nodoc:
+    def with_output_buffer(buf = "".dup) #:nodoc:
+      raise "No: #{buf}" if buf.frozen?
```

Comment:
Maybe a better descriptive error here. Also I think the exception should be an ArgumentError